test.cmake

```diff
@@ -18,19 +18,19 @@
 #
 #   Boolean variable to configure unit test compilation declared with
 #   :command:`ADD_UNIT_TEST`.
+
+#   A target *build_tests* is added to compile the tests and
+#   a test that run target build_tests is run before all other tests.
+#   So command ``make test`` compiles and runs the unit-tests.
 #
 #   * if ``OFF`` (default), nothing special is done.
 #   * if ``ON``, the unit-test is not compiled with target *all*.
-#     A target *build_tests* is added to compile the tests and
-#     a test that run target build_tests is run before all other tests.
-#     So command ``make test`` compiles and runs the unit-tests.
 IF(NOT DEFINED DISABLE_TESTS)
   SET(DISABLE_TESTS OFF)
 ENDIF(NOT DEFINED DISABLE_TESTS)
-IF(DISABLE_TESTS)
-  ADD_TEST(ctest_build_tests "${CMAKE_COMMAND}" --build ${CMAKE_BINARY_DIR} --target build_tests)
-  ADD_CUSTOM_TARGET(build_tests)
-ENDIF(DISABLE_TESTS)
+
+ADD_CUSTOM_TARGET(build_tests)
+ADD_TEST(ctest_build_tests "${CMAKE_COMMAND}" --build ${CMAKE_BINARY_DIR} --target build_tests)
```

Comment:
I don't think we wan't to add test `ctest_build_tests` when DISABLE_TESTS is OFF. This target is a hack to bypass the lack of versatility of target test. I propose this instead:
```cmake
IF(DISABLE_TESTS)
   ADD_TEST(ctest_build_tests "${CMAKE_COMMAND}" --build ${CMAKE_BINARY_DIR} --target build_tests)
ENDIF()
ADD_CUSTOM_TARGET(build_tests)
```

Reply:
I realize that this is the only known solution to compile tests by calling `make test` without creating additional target like `make check`.

Reply:
We should not modify the default behaviour of target test to avoid surprises to people. If you want to compile and run the test, you must run `make build_tests && make test`. This is the standard imposed by CMake and I think we should stick with it.

Now, coming to the hack of compiling the tests when a user does `make test` with DISABLE_TESTS enabled: the purpose is mostly to avoid stupid errors like "test failed" while you asked not to compile them but to run them...

Reply:
So to conclude, when DISABLE_TESTS is disabled, `make test` should stick to CMake standard and not compile the unit tests.

Reply:
Eventually, we can add a target `check`. I just prefer not to modify `test`.

Reply:
I will do the modifications for supporting it.